### PR TITLE
LocalView redesigned, logging improvements, cleanup of consensus instances.

### PR DESCRIPTION
**Pull request by kape1395**

### CODE REVIEW

File: packages/chain/cmtLog/var_localView.go [hunk 1]
```diff
@@ -28,123 +78,254 @@ type VarLocalView interface {
...
+	pendingSICount := uint32(lvi.pending.Size())
+	for i := uint32(0); i < pendingSICount; i++ {
+		entries, ok := lvi.pending.Get(confirmedSI + i + 1)
+		if !ok {
+			return nil // That's a gap.
+		}
+		if len(entries) != 1 {
+			return nil // Alternatives exist.
+		}
+		if entries[0].rejected {
```
Comment (muXxer) [inline packages/chain/cmtLog/var_localView.go:299]: Do we need to loop over all entries here just to be safe that none of them is rejected? Or is this already clear if we only check the first entry?

File: packages/chain/cmtLog/var_localView.go [hunk 2]
```diff
@@ -9,16 +9,66 @@
...
+//     be rejected eventually as well, because they use the rejected AO as an input. On the
+//     other hand, it is unclear if unconfirmed AOs before the rejected one will be confirmed
+//     or rejected, we will wait until L1 decides on all of them.
+//
+//   - If we get a confirmed AO, that is not one of the AOs we have posted (and still waiting
+//     for a decision), then someone from the outside of a committee transitioned the chain.
+//     In this case all our produced/pending transactions are not meaningful anymore and we
+//     have to start building all the chain from the newly received AO.
+//
+//   - Recovery notice is received from a consensus (with SI/LI...) a new new consensus will
```
Comment (muXxer) [inline packages/chain/cmtLog/var_localView.go]:
```suggestion
//   - Recovery notice is received from a consensus (with SI/LI...) a new consensus will
```

File: packages/chain/cmtLog/var_localView.go (hunk 1 above)
Comment (kape1395) [reply packages/chain/cmtLog/var_localView.go:299]: The check above tests if there is only a single element in the entries list. So it is enough here to check that single element.

Comment (muXxer) [reply packages/chain/cmtLog/var_localView.go:299]: hahaha true :D sorry my bad.

**Review by muXxer — APPROVED** [review]

**Pull request merged**